pkg/controller/batch/computeenvironment/setup.go

```diff
@@ -0,0 +1,553 @@
+	fmt.Printf("in: %v\n", in)
+
+	if _, err := e.client.UpdateComputeEnvironment(in); err != nil {
+		return true, awsclients.Wrap(err, "UpdateComputeEnvironment-call for CE deletion failed")
+	} */ //leaving this here in case the way below makes problems again (not sure if it was a one time thing, misinterpreted or what)
+
+	//thought this could be faster for EC2 type Deletion, however unsure if the updating of vCPUs really ...
+	//CB: Test: only if time (as it needs to be tested for all types)
+	/* in := &svcsdk.UpdateComputeEnvironmentInput{
+		ComputeEnvironment: awsclients.String(meta.GetExternalName(cr)),
+		State:              awsclients.String(svcsdk.CEStateDisabled)}
+	//MANAGED EC2 ... SPOT too?
+	if cr.Spec.ForProvider.Type == awsclients.String(svcsdk.CETypeManaged) && cr.Spec.ForProvider.ComputeResources.Type == awsclients.String(svcsdk.CRTypeEc2) {
+		in.ComputeResources.SetMinvCpus(0) //desired needs to be between min and max
+		in.ComputeResources.SetDesiredvCpus(0)
+	}
+	if _, err : = e.client.UpdateComputeEnvironment(in) {
+		return true, awsclients.Wrap(err, "UpdateComputeEnvironment-call for CE deletion failed")
+	} */ //Alternative to make deletion maybe faster for EC2
+
+	_, err := e.client.UpdateComputeEnvironment(&svcsdk.UpdateComputeEnvironmentInput{
+		ComputeEnvironment: awsclients.String(meta.GetExternalName(cr)),
+		State:              awsclients.String(svcsdk.CEStateDisabled)})
+	return true, awsclients.Wrap(err, "UpdateComputeEnvironment-call for CE deletion failed")
+	//depending on the CE Setup (type, compute resources, status ...) Update-Call can take some while (-> status UPDATING) before getting to the Deletion...
+}
+
+func postDelete(_ context.Context, cr *svcapitypes.ComputeEnvironment, obj *svcsdk.DeleteComputeEnvironmentOutput, err error) error {
+
+	//CB: possibly not needed when predelete solves all...
+	if err != nil {
+		if strings.Contains(err.Error(), "modified") {
+			// skip: Can't delete workspace in non-ACTIVE state. Current status is DELETING
+
+			fmt.Println("postDelete skipped err modified ", err)
+			return nil
+		}
+		return err
+	}
+	return err
+}
+
+func (e *hooks) isUpToDate(cr *svcapitypes.ComputeEnvironment, obj *svcsdk.DescribeComputeEnvironmentsOutput) (bool, error) {
+
+	fmt.Println("isUpToDate: cr.Status.AtProvider.Status :", awsclients.StringValue(cr.Status.AtProvider.Status))
+	// Skip when updating, deleting or creating
+	if awsclients.StringValue(cr.Status.AtProvider.Status) == string(svcapitypes.CEStatus_UPDATING) ||
+		awsclients.StringValue(cr.Status.AtProvider.Status) == string(svcapitypes.CEStatus_DELETING) ||
+		awsclients.StringValue(cr.Status.AtProvider.Status) == string(svcapitypes.CEStatus_CREATING) {
+
+		return true, nil
+	}
+
+	ce := obj.ComputeEnvironments[0]
+
+	// check for changes in CE-Tags
+	if !cmp.Equal(cr.Spec.ForProvider.Tags, ce.Tags, cmpopts.EquateEmpty()) {
+
+		err := e.UpdateCETags(cr, ce)
```

Comment:
Tags must_not_ be updated in `isUpToDate` but in `postUpdate` instead. In general `Observe` should be read-only.